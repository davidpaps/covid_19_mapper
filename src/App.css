.App {
  text-align: center;
  background-color: rgb(81, 82, 82);
}

.Container {
  display: flex;
  flex-direction: row;
}

#drop-down-window {
  font-size: 14px;
  color: grey;
  font-weight: bolder;
}

#map {
  position: absolute;
  width: 100%;
  height: 90%;
  right: 0px;
}

#legend {
  position: absolute;
  display: flex;
  flex-direction: column;
  z-index: 1;
  top: 350px;
  color: white;
  left: 0px;
  width: 5%;
}

#mapinfo {
  position: absolute;
  display: flex;
  flex-direction: row;
  width: 100%;
  z-index: 1;
  bottom: 0px;
  fill: none;
}

#info {
  width: 15%;
}

#infoUS {
  width: 40%;
}