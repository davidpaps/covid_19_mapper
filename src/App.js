import React, { Component } from "react";
import "./App.css";
import Header from "./components/navbar/navbar";
import ref_country_codes from "./components/assets/countries-lat-long.json";
import us_codes from "./components/assets/USlatlong.json";

class App extends Component {
  state = {
    countries: [],
    error: null,
    isLoaded: false,
    total: [],
    totalInt: [],
    totalCFR: null,
    countriesInteger: [],
  };

  componentDidMount() {
    Promise.all([
      fetch(
        "https://coronavirus-monitor.p.rapidapi.com/coronavirus/worldstat.php",
        {
          method: "GET",
          headers: {
            "x-rapidapi-host": "coronavirus-monitor.p.rapidapi.com",
            "x-rapidapi-key":
              "2bb49386fdmsh5daac6ca9add22ep1484a8jsn9816903163ef",
          },
        }
      ),
      fetch(
        "https://coronavirus-monitor.p.rapidapi.com/coronavirus/cases_by_country.php",
        {
          method: "GET",
          headers: {
            "x-rapidapi-host": "coronavirus-monitor.p.rapidapi.com",
            "x-rapidapi-key":
              // "2bb49386fdmsh5daac6ca9add22ep1484a8jsn9816903163ef"
              "a04279196bmsh77bb3ff9e6f2e74p1f4d03jsn5bc0fbe15879",
          },
        }
      ),
      fetch("https://covid19-data.p.rapidapi.com/us", {
        method: "GET",
        headers: {
          "x-rapidapi-host": "covid19-data.p.rapidapi.com",
          "x-rapidapi-key":
            "2bb49386fdmsh5daac6ca9add22ep1484a8jsn9816903163ef",
        },
      }),
    ])

      .then(([res1, res2, res3]) => {
        return Promise.all([res1.json(), res2.json(), res3.json()]);
      })

      .then(([res1, res2, res3]) => {
        this.setState({
          countries: this.createCountry(res2.countries_stat, res3.list),
          total: this.updateTotal(res1),
          totalInt: this.toInteger(res1),
          totalCFR: this.makeGlobalCFR(res2.countries_stat, res3.list),
          countriesInteger: this.makeCountriesInteger(
            res2.countries_stat,
            res3.list
          ),
        });
      });
  }

  createCountry(country, states) {
    const countries = [];
    ref_country_codes.ref_country_codes.forEach((one) =>
      country.forEach((two) => {
        if (two.country_name === "UK") {
          two.country_name = "United Kingdom";
        }
        if (one.country === two.country_name) {
          let intTotalConfirmed = parseInt(two.cases.replace(/,/g, ""));
          let intTotalDeaths = parseInt(two.deaths.replace(/,/g, ""));
          let cfr = (intTotalDeaths / intTotalConfirmed) * 100;
          countries.push({
            country: two.country_name,
            recovered: two.total_recovered,
            deaths: two.deaths,
            confirmed: two.cases,
            center: { lat: one.latitude, lng: one.longitude },
            newCases: two.new_cases,
            newDeaths: two.new_deaths,
            activeCases: two.active_cases,
            criticalCases: two.serious_critical,
            perOneMillion: two.total_cases_per_1m_population,
            cfr: parseFloat(cfr.toFixed(2)),
          });
        }
      })
    );

    let a = this.updateUS(states, countries);

    return a;
  }

  updateUS(states, countries) {
    us_codes.us_codes.forEach((state) =>
      states.forEach((obj) => {
        if (obj.state === "Georgia") {
          obj.state = "Georgia, US";
        }
        if (obj.state === state.state) {
          countries.push({
            us: true,
            country: state.state,
            recovered: obj.recovered,
            deaths: this.commaNumberString(obj.deaths.toString()),
            confirmed: this.commaNumberString(obj.confirmed.toString()),
            center: { lat: state.latitude, lng: state.longitude },
            cfr: parseFloat(((obj.deaths / obj.confirmed) * 100).toFixed(2)),
          });
        }
      })
    );
    return countries;
  }

  updateTotal(totalArray) {
    let total = this.toInteger(totalArray);
    var activeCases = total[0] - total[1] - total[2];

    totalArray["active_cases"] = this.commaNumberString(activeCases.toString())
    return totalArray;
  }

  makeGlobalCFR(countries, states) {
    let cfrPerCountry = [];
    countries.forEach((country) => {
      cfrPerCountry.push(
        (parseInt(country.deaths.replace(/,/g, "")) /
          parseInt(country.cases.replace(/,/g, ""))) *
          100
      );
    });

    states.forEach((state) => {
      cfrPerCountry.push((state.deaths / state.confirmed) * 100);
    });
    const reducer = (accumulator, currentValue) => accumulator + currentValue;
    let avCfr = cfrPerCountry.reduce(reducer) / cfrPerCountry.length;

    return parseFloat(avCfr.toFixed(2));
  }

  toInteger(totalArray) {
    var newTotalCases = parseInt(totalArray["total_cases"].replace(/,/g, ""));
    var newTotalDeaths = parseInt(totalArray["total_deaths"].replace(/,/g, ""));
    var newTotalRecoveries = parseInt(
      totalArray["total_recovered"].replace(/,/g, "")
    );
    return [newTotalCases, newTotalDeaths, newTotalRecoveries];
  }

  makeCountriesInteger(countries, states) {
    const countriesInteger = [];

    ref_country_codes.ref_country_codes.forEach((one) => {
      countries.forEach((two) => {
        if (one.country === two.country_name) {
          let intTotalConfirmed = parseInt(two.cases.replace(/,/g, ""));
          let intTotalDeaths = parseInt(two.deaths.replace(/,/g, ""));
          let cfr = (intTotalDeaths / intTotalConfirmed) * 100;

          countriesInteger.push({
            country: two.country_name,
            recovered: parseInt(two.total_recovered.replace(/,/g, "")),
            deaths: parseInt(two.deaths.replace(/,/g, "")),
            confirmed: parseInt(two.cases.replace(/,/g, "")),
            center: { lat: one.latitude, lng: one.longitude },
            newCases: parseInt(two.new_cases.replace(/,/g, "")),
            newDeaths: parseInt(two.new_deaths.replace(/,/g, "")),
            activeCases: parseInt(two.active_cases.replace(/,/g, "")),
            criticalCases: parseInt(two.serious_critical.replace(/,/g, "")),
            perOneMillion: parseInt(
              two.total_cases_per_1m_population.replace(/,/g, "")
            ),
            cfr: parseFloat(cfr.toFixed(2)),
          });
        }
      });
    });

    us_codes.us_codes.forEach((state) => {
      states.forEach((obj) => {
        if (obj.state === "Georgia") {
          obj.state = "Georgia, US";
        }
        if (obj.state === state.state) {
          countriesInteger.push({
            us: true,
            country: state.state,
            recovered: obj.recovered,
            deaths: obj.deaths,
            confirmed: obj.confirmed,
            center: { lat: state.latitude, lng: state.longitude },
            cfr: parseFloat(((obj.deaths / obj.confirmed) * 100).toFixed(2)),
          });
        }
      });
    });
    return countriesInteger;
  }

  commaNumberString(string) {
    return string.replace(/\d{1,3}(?=(\d{3})+(?!\d))/g , "$&,");
  }

  render() {
    return (
      <div className="App">
        <Header
          total={this.state.total}
          totalInt={this.state.totalInt}
          countries={this.state.countries}
          globalCFR={this.state.totalCFR}
          integerCountries={this.state.countriesInteger}
        />
      </div>
    );
  }
}

export default App;
