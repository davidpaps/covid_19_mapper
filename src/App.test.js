import * as React from 'react';
import { render } from '@testing-library/react';
import { shallow } from 'enzyme';
import App from './App';
import renderer from 'react-test-renderer';



describe("App Component", () => {


  beforeEach(() => {
    shallow(<App />, { disableLifeCycleMethods: true })
  })

  it('renders as expected, snapshot comparison', () => {
    const TextInputComponent = renderer.create(<App />).toJSON()
    expect(TextInputComponent).toMatchSnapshot()
  })

  it('mounts the state', () => {
    const spy = jest.spyOn(App.prototype, 'componentDidMount')
    shallow(<App />)
    expect(spy).toHaveBeenCalled()
  })

  it('calls a fetch', () => {
    const fetchSpy = jest.spyOn(window, "fetch")
    shallow(<App />)
    expect(fetchSpy).toHaveBeenCalled()
  })
})

describe('createCountry', () => {
  var app = new App()
  const countries = [
    {
      country_name: "USA",
      cases: "338,899",
      deaths: "9,679",
      region: "",
      total_recovered: "18,002",
      new_deaths: "63",
      new_cases: "2,226",
      serious_critical: "8,702",
      active_cases: "311,218",
      total_cases_per_1m_population: "1,024",
    }]
  const state = [
    {
      countrycode: "US",
      country: "United States of America",
      state: "South Carolina",
      latitude: "34.22333378",
      longitude: "-82.46170658",
      confirmed: 2049,
      deaths: 44,
    }
  ]
  it("creates the country array", () => {
    expect(app.createCountry(countries, state)).toEqual([
      {
        country: "USA",
        recovered: "18,002",
        deaths: "9,679",
        confirmed: "338,899",
        center: { lat: 45.301456, lng: -104.086992 },
        newCases: "2,226",
        newDeaths: "63",
        activeCases: "311,218",
        criticalCases: "8,702",
        perOneMillion: "1,024",
        cfr: "2.86"
      },
      {
        us: true,
        country: "South Carolina",
        recovered: undefined,
        deaths: "44",
        confirmed: "2,049",
        center: { lat: 33.8191, lng: -80.9066 },
        cfr: "2.15",
      }])
  })

})

describe("updateUS", () => {
  var app = new App()
  const countries = [
    {
      country: "USA",
      recovered: "18,002",
      deaths: "9,679",
      confirmed: "338,899",
      center: { lat: 35.222, lng: -101.8313 },
      newCases: "2,226",
      newDeaths: "63",
      activeCases: "311,218",
      criticalCases: "8,702",
      perOneMillion: "1,024",
    }]
  const states = [
    {
      countrycode: "US",
      country: "United States of America",
      state: "Georgia",
      latitude: "31.74847232",
      longitude: "-82.28909114",
      confirmed: 6647,
      deaths: 211
    }]

  it("creates the Georgia correctly", () => {
    expect(app.updateUS(states, countries)).toEqual([
      {
        country: "USA",
        recovered: "18,002",
        deaths: "9,679",
        confirmed: "338,899",
        center: { lat: 35.222, lng: -101.8313 },
        newCases: "2,226",
        newDeaths: "63",
        activeCases: "311,218",
        criticalCases: "8,702",
        perOneMillion: "1,024",
      },
      {
        us: true,
        country: "Georgia, US",
        recovered: undefined,
        deaths: "211",
        confirmed: "6,647",
        center: { lat: 32.9866, lng: -83.6487 },
        cfr: "3.17",
      }])
  })

})

describe("updateTotal", () => {
  var app = new App()
  const total = {
    total_cases: "1,250,000",
    total_deaths: "70,000",
    total_recovered: "270,000",
    new_cases: "14,000",
    new_deaths: "1000",
    statistic_taken_at: "2020-04-06 12:53:08"
  }
  it("adds the active cases to the object", () => {
    expect(app.updateTotal(total)).toEqual({
      total_cases: "1,250,000",
      total_deaths: "70,000",
      total_recovered: "270,000",
      new_cases: "14,000",
      new_deaths: "1000",
      statistic_taken_at: "2020-04-06 12:53:08",
      active_cases: "910,000"
    })
  })
})