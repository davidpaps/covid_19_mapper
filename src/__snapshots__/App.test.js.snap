// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders as expected, snapshot comparison 1`] = `
<div
  className="App"
>
  <nav
    className="navbar-dark bs-navbar-collapse navbar navbar-expand navbar-light fixed-top"
  >
    <span
      className="navbar-brand"
    >
      <img
        alt=""
        className="d-inline-block align-top"
        height="30"
        src="https://cdn3.iconfinder.com/data/icons/science-116/64/virus-lab-scientist-biology-cell-medical-512.png?v=2"
        width="30"
      />
       
      Covid-19
    </span>
    <div
      className="dropdown"
      onKeyDown={[Function]}
    >
      <button
        aria-expanded={false}
        aria-haspopup={true}
        className="dropdown-toggle btn btn-outline-warning"
        disabled={false}
        id="last_updated"
        onClick={[Function]}
        type="button"
      >
        <span
          aria-hidden="true"
          className="spinner-grow spinner-grow-sm"
          role="status"
        />
      </button>
    </div>
    <div
      className="ml-auto navbar-nav"
      onKeyDown={[Function]}
    >
      <div
        className="dropdown"
        onKeyDown={[Function]}
      >
        <button
          aria-expanded={false}
          aria-haspopup={true}
          className="dropdown-toggle btn btn-outline-success"
          disabled={false}
          id="last_updated"
          onClick={[Function]}
          type="button"
        >
          <span
            aria-hidden="true"
            className="spinner-grow spinner-grow-sm"
            role="status"
          />
        </button>
      </div>
    </div>
    <div
      className="ml-auto navbar-nav"
      onKeyDown={[Function]}
    >
      <div
        className="dropdown"
        onKeyDown={[Function]}
      >
        <button
          aria-expanded={false}
          aria-haspopup={true}
          className="dropdown-toggle btn btn-outline-danger"
          disabled={false}
          id="last_updated"
          onClick={[Function]}
          type="button"
        >
          <span
            aria-hidden="true"
            className="spinner-grow spinner-grow-sm"
            role="status"
          />
        </button>
      </div>
    </div>
    <div
      className="ml-auto navbar-nav"
      onKeyDown={[Function]}
    >
      <div
        className="dropdown"
        onKeyDown={[Function]}
      >
        <button
          aria-expanded={false}
          aria-haspopup={true}
          className="dropdown-toggle btn btn-outline-info"
          disabled={false}
          id="dropdown-menu-align-right"
          onClick={[Function]}
          type="button"
        >
          <span
            aria-hidden="true"
            className="spinner-grow spinner-grow-sm"
            role="status"
          />
        </button>
      </div>
    </div>
  </nav>
  <div
    className="Container"
  >
    <div>
      <div>
        <div
          id="container"
        >
          <div
            id="map"
          />
          <div />
        </div>
      </div>
      <div
        id="legend"
      >
        <div>
          <div
            id="key"
          >
            <strong>
              Key
            </strong>
            <br />
            <button
              className="btn btn-outline- btn-sm"
              disabled={false}
              onBlur={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              type="button"
            >
              <div
                id="colour"
              >
                <svg
                  height="30"
                  styles=" fill:#000000;"
                  viewBox="0 0 172 172"
                  width="30"
                  x="0px"
                  xmlns="http://www.w3.org/2000/svg"
                  y="0px"
                >
                  <g
                    fill="none"
                    fill-rule="nonzero"
                  >
                    <path
                      d="M0,172v-172h172v172z"
                      fill="none"
                    />
                    <g
                      fill="#ffc108"
                    >
                      <path
                        d="M86,157.66667c-39.41667,0 -71.66667,-32.25 -71.66667,-71.66667c0,-39.41667 32.25,-71.66667 71.66667,-71.66667c39.41667,0 71.66667,32.25 71.66667,71.66667c0,39.41667 -32.25,71.66667 -71.66667,71.66667z"
                      />
                    </g>
                  </g>
                </svg>
              </div>
            </button>
            <br />
            <button
              className="btn btn-outline- btn-sm"
              disabled={false}
              onBlur={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              type="button"
            >
              <div
                id="colour"
              >
                <svg
                  height="30"
                  styles=" fill:#000000;"
                  viewBox="0 0 172 172"
                  width="30"
                  x="0px"
                  xmlns="http://www.w3.org/2000/svg"
                  y="0px"
                >
                  <g
                    fill="none"
                    fill-rule="nonzero"
                  >
                    <path
                      d="M0,172v-172h172v172z"
                      fill="none"
                    />
                    <g
                      fill="#dc3645"
                    >
                      <path
                        d="M86,157.66667c-39.41667,0 -71.66667,-32.25 -71.66667,-71.66667c0,-39.41667 32.25,-71.66667 71.66667,-71.66667c39.41667,0 71.66667,32.25 71.66667,71.66667c0,39.41667 -32.25,71.66667 -71.66667,71.66667z"
                      />
                    </g>
                  </g>
                </svg>
              </div>
            </button>
          </div>
          <button
            className="btn btn-outline- btn-sm"
            disabled={false}
            onBlur={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            type="button"
          >
            <div
              id="size"
            >
              <svg
                height="15"
                styles=" fill:#000000;"
                viewBox="0 0 172 172"
                width="15"
                x="0px"
                xmlns="http://www.w3.org/2000/svg"
                y="0px"
              >
                <g
                  transform="translate(4.988,4.988) scale(0.942,0.942)"
                >
                  <g
                    fill="none"
                    fill-rule="nonzero"
                    s={true}
                  >
                    <g
                      fill="#ffffff"
                      stroke="#cccccc"
                      stroke-linejoin="round"
                      stroke-width="10"
                    >
                      <path
                        d="M86,157.66667c-39.41667,0 -71.66667,-32.25 -71.66667,-71.66667c0,-39.41667 32.25,-71.66667 71.66667,-71.66667c39.41667,0 71.66667,32.25 71.66667,71.66667c0,39.41667 -32.25,71.66667 -71.66667,71.66667zM86,28.66667c-31.53333,0 -57.33333,25.8 -57.33333,57.33333c0,31.53333 25.8,57.33333 57.33333,57.33333c31.53333,0 57.33333,-25.8 57.33333,-57.33333c0,-31.53333 -25.8,-57.33333 -57.33333,-57.33333z"
                      />
                    </g>
                    <path
                      d="M0,172v-172h172v172z"
                      fill="none"
                      stroke="none"
                      stroke-linejoin="miter"
                      stroke-width="1"
                    />
                    <g
                      fill="#ffffff"
                      stroke="none"
                      stroke-linejoin="miter"
                      stroke-width="1"
                    >
                      <path
                        d="M86,157.66667c-39.41667,0 -71.66667,-32.25 -71.66667,-71.66667c0,-39.41667 32.25,-71.66667 71.66667,-71.66667c39.41667,0 71.66667,32.25 71.66667,71.66667c0,39.41667 -32.25,71.66667 -71.66667,71.66667zM86,28.66667c-31.53333,0 -57.33333,25.8 -57.33333,57.33333c0,31.53333 25.8,57.33333 57.33333,57.33333c31.53333,0 57.33333,-25.8 57.33333,-57.33333c0,-31.53333 -25.8,-57.33333 -57.33333,-57.33333z"
                      />
                    </g>
                    <path
                      d=""
                      fill="none"
                      stroke="none"
                      stroke-linejoin="miter"
                      stroke-width="1"
                    />
                  </g>
                </g>
              </svg>
            </div>
          </button>
          <br />
          <button
            className="btn btn-outline- btn-sm"
            disabled={false}
            onBlur={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            type="button"
          >
            <div
              id="size"
            >
              <svg
                height="30"
                styles=" fill:#000000;"
                viewBox="0 0 172 172"
                width="30"
                x="0px"
                xmlns="http://www.w3.org/2000/svg"
                y="0px"
              >
                <g
                  transform="translate(4.988,4.988) scale(0.942,0.942)"
                >
                  <g
                    fill="none"
                    fill-rule="nonzero"
                    s={true}
                  >
                    <g
                      fill="#ffffff"
                      stroke="#cccccc"
                      stroke-linejoin="round"
                      stroke-width="10"
                    >
                      <path
                        d="M86,157.66667c-39.41667,0 -71.66667,-32.25 -71.66667,-71.66667c0,-39.41667 32.25,-71.66667 71.66667,-71.66667c39.41667,0 71.66667,32.25 71.66667,71.66667c0,39.41667 -32.25,71.66667 -71.66667,71.66667zM86,28.66667c-31.53333,0 -57.33333,25.8 -57.33333,57.33333c0,31.53333 25.8,57.33333 57.33333,57.33333c31.53333,0 57.33333,-25.8 57.33333,-57.33333c0,-31.53333 -25.8,-57.33333 -57.33333,-57.33333z"
                      />
                    </g>
                    <path
                      d="M0,172v-172h172v172z"
                      fill="none"
                      stroke="none"
                      stroke-linejoin="miter"
                      stroke-width="1"
                    />
                    <g
                      fill="#ffffff"
                      stroke="none"
                      stroke-linejoin="miter"
                      stroke-width="1"
                    >
                      <path
                        d="M86,157.66667c-39.41667,0 -71.66667,-32.25 -71.66667,-71.66667c0,-39.41667 32.25,-71.66667 71.66667,-71.66667c39.41667,0 71.66667,32.25 71.66667,71.66667c0,39.41667 -32.25,71.66667 -71.66667,71.66667zM86,28.66667c-31.53333,0 -57.33333,25.8 -57.33333,57.33333c0,31.53333 25.8,57.33333 57.33333,57.33333c31.53333,0 57.33333,-25.8 57.33333,-57.33333c0,-31.53333 -25.8,-57.33333 -57.33333,-57.33333z"
                      />
                    </g>
                    <path
                      d=""
                      fill="none"
                      stroke="none"
                      stroke-linejoin="miter"
                      stroke-width="1"
                    />
                  </g>
                </g>
              </svg>
            </div>
          </button>
        </div>
      </div>
      <div
        id="mapinfo"
      >
        <div
          className="bg-warning text-white"
          id="card"
        />
      </div>
    </div>
  </div>
</div>
`;
